crypto/zkp/schemes/cl/user.go

```diff
@@ -26,22 +26,45 @@ import (
 )
 
 type User struct {
-	ParamSizes     *CLParamSizes
-	PubKey         *PubKey
-	PedersenParams *commitments.PedersenParams               // for pseudonyms - nym is a commitment to the master secret
-	Committers     map[string]*commitments.PedersenCommitter // for generating nyms
-	masterSecret   *big.Int
-	attrs          []*big.Int
+	ParamSizes         *CLParamSizes
+	PubKey             *PubKey
+	PedersenParams     *commitments.PedersenParams               // for pseudonyms - nym is a commitment to the master secret
+	Committers         map[string]*commitments.PedersenCommitter // for generating nyms
+	masterSecret       *big.Int
+	knownAttrs         []*big.Int                              // attributes that are known to the credential receiver and issuer
+	committedAttrs     []*big.Int                              // attributes for which the issuer knows only commitment
+	hiddenAttrs        []*big.Int                              // attributes which are known only to the credential receiver
+	attrsCommitters    []*commitments.DamgardFujisakiCommitter // committers for committedAttrs
+	commitmentsOfAttrs []*big.Int                              // commitments of committedAttrs
 }
 
-func NewUser(clParamSizes *CLParamSizes, clPubKey *PubKey, pedersenParams *commitments.PedersenParams) *User {
-	return &User{
-		ParamSizes:     clParamSizes,
-		PubKey:         clPubKey,
-		Committers:     make(map[string]*commitments.PedersenCommitter),
-		PedersenParams: pedersenParams,
-		attrs:          []*big.Int{big.NewInt(7), big.NewInt(6), big.NewInt(5)}, // TODO attributes should be read from somewhere and the structure should be loaded too
+func NewUser(clParamSizes *CLParamSizes, clPubKey *PubKey, pedersenParams *commitments.PedersenParams,
```

Comment:
A question - are the values of user's `CLParamSizes` in any way connected to known/committed/hidden attributes?

I was thinking that, lets say, I set some values of `CLParamSizes`, but then pass some attributes which do not conform to some of these values from `CLParamSizes`. Can this happen? If it can, we need to introduce some checks to see if parameter values are consistent with the rest of arguments we're getting in the `NewUser` function.